pkg/cmd/service/service_log.go

```diff
@@ -0,0 +1,161 @@
+				opts := logs.Options{
+					Follow: follow,
+					Format: output,
+					Skip:   -1 * lines,
+				}
+				return logClient.PrintComponent("/leader/mesos", "/dcos-marathon.service", opts)
+			}
+
+			appID, err := serviceAppID(marathonClient, args[0])
+			if err != nil {
+				return err
+			}
+			taskID, err := serviceTaskID(marathonClient, appID)
+			if err != nil {
+				return err
+			}
+
+			logClient := logs.NewClient(pluginutil.HTTPClient(""), ctx.Out())
+			opts := logs.Options{
+				Follow: follow,
+				Format: output,
+				Skip:   -1 * lines,
+			}
+			return logClient.PrintTask(taskID, file, opts)
+		},
+	}
+
+	cmd.Flags().BoolVar(&follow, "follow", false, "Dynamically update the log")
+	cmd.Flags().IntVar(&lines, "lines", 10, "Dynamically update the log")
+	cmd.Flags().StringVarP(&output, "output", "o", "short", "Format log message output")
+	cmd.Flags().String("user", "", "The SSH user for Marathon")
+	cmd.Flags().String("ssh-config-file", "", "Path to SSH configuration file. This is depracated.")
+	return cmd
+}
+
+func marathonMasterURL(c *marathon.Client, ctx api.Context) (string, error) {
+	ipsRes := make(chan ipsResult)
+	go publicIPs(ipsRes)
+
+	leader, err := c.API.Leader()
+	if err != nil {
+		return "", err
+	}
+
+	publicIPs := <-ipsRes
+	if publicIPs.err != nil {
+		return "", nil
+	}
+
+	// from <ip>:<port> get the <ip> field
+	marathonIP := strings.Split(leader, ":")[0]
+
+	cluster, err := ctx.Cluster()
+	if err != nil {
+		return "", err
+	}
+	// from <scheme>://<url> get the <scheme> field
+	scheme := strings.Split(cluster.URL(), "://")[0]
+
+	return fmt.Sprintf("%s://%s", scheme, publicIPs.ips[marathonIP][0]), nil
+}
+
+func publicIPs(ch chan<- ipsResult) {
+	c := networking.NewClient(pluginutil.HTTPClient(""))
+	nodes, err := c.Nodes()
+
+	ips := make(map[string][]string)
+	if err == nil {
+		for _, node := range nodes {
+			ips[node.PrivateIP] = node.PublicIPs
+		}
+	}
+	ch <- ipsResult{ips, err}
+}
+
+func serviceAppID(c *marathon.Client, service string) (string, error) {
+	apps, err := c.API.Applications(nil)
+	if err != nil {
+		return "", err
+	}
+
+	var appIDs []string
+	for _, a := range apps.Apps {
+		if (*a.Labels)["DCOS_PACKAGE_FRAMEWORK_NAME"] == service {
+			appIDs = append(appIDs, a.ID)
+		}
+	}
+
+	switch len(appIDs) {
+	case 0:
+		return "", fmt.Errorf("No marathon apps found for service name [%s]", service)
+	case 1:
+		return appIDs[0], nil
+	default:
+		return "", fmt.Errorf("Multiple marathon apps found for service name [%s]: [%s]", service, strings.Join(appIDs,", "))
+	}
+}
+
+func serviceTaskID(c *marathon.Client, appID string) (string, error) {
+	app, err := c.API.Application(appID)
+	if err != nil {
+		return "", err
+	}
+
+	switch len(app.Tasks) {
+	case 1:
+		return app.Tasks[0].ID , nil
+	default:
+		return "", fmt.Errorf("We expected marathon app [%s] to be running 1 task, but we instead found %d tasks", appID, len(app.Tasks))
```

Comment:
We rarely use "We" when logging something, we can remove it.

Reply:
Already changed it, that was copied from the messages of the old code.